Added blue color to tanks

pridhviu opened the pull request:
I want to resolve #5 by adding BLUE colour to the tanks while rendering main playing screen.

pridhviu commented:
Added collision detection and boundary checking to resolve #9

thuanpv commented:
Hi @pridhvi. Thank you for your first PR. Regarding the changes to add colors for the tanks, I expect a more generic solution that can apply to tanks as well as other objects on the playfield. I was thinking of adding a color attribute.

Regarding the collision detection algorithm, it would be great if you could present your solution to your class mates in the live lecture tomorrow.

pridhviu commented:
Hi @thuanpv I have done the following,

- Removed the blue colour printing in the previous commit.
- Added a new class Object which acts as the parent class for all objects in the game map including Tanks and other obstacles. I have made Tank class to inherit from Object class. This object class has a color attribute which can used when implementing the color render.

essut commented on CityTankGame.java:
```diff
@@ -34,7 +34,7 @@ public static void main(String[] args)
       //Get user command
       System.out.println("Control your tank by typing a character and press Enter. ");
-      System.out.println("W: up, A: left, S: right, Z: down, Q: quit the game.");
+      System.out.println("W: up, A: left, D: right, S: down, Q: quit the game.");
```
I don't think this needs to be changed. It is valid to use a non-WASD control.

pridhviu replied:
I have reverted it.

essut commented on objects/Object.java:
```diff
@@ -0,0 +1,54 @@
+package objects;
```
This is implementing the game using inheritance. Can you kindly split the pull requests so it is easier to review?

pridhviu replied:
I have created a separate PR for this. #12

essut commented on objects/Tank.java:
```diff
@@ -57,10 +31,10 @@ public void move(Commands.Direction cmd)
         {
           case UP:
-            rowPos--;
+            if(! isCollision(mainRenderer, getRowPos() - 1, getColPos())) setRowPos(getRowPos() - 1);
```
I don't prefer this kind of if statement. Wouldn't it be better if the condition and statement is split into separate lines?

pridhviu replied:
I have made the if condition into multiple lines

essut commented on objects/Tank.java:
```diff
@@ -171,4 +141,25 @@ public void move(Commands.Direction cmd)
   }
-}
+
+  public boolean isCollision(Renderer mainRenderer, int rowPos, int colPos) {
```
I think this is a helper function, which should be private.

pridhviu replied:
Thanks for the suggestion I missed that. I have made it private now

thuanpv commented:
I close this PR because the newer one has been merged. Thanks you both! Keep up the good work!
